pkg/materialize/connection_kafka.go

```diff
@@ -38,19 +38,51 @@ func GetKafkaBrokersStruct(v interface{}) []KafkaBroker {
 	return brokers
 }
 
+type awsPrivateLinkConnection struct {
+	PrivateLinkConnection IdentifierSchemaStruct
+	PrivateLinkPort       int
+}
+
+func GetAwsPrivateLinkConnectionStruct(v interface{}) awsPrivateLinkConnection {
+	if v == nil || len(v.([]interface{})) == 0 {
+		return awsPrivateLinkConnection{}
+	}
+
+	plMap, ok := v.([]interface{})[0].(map[string]interface{})
+	if !ok {
+		return awsPrivateLinkConnection{}
+	}
+
+	privatelinkConnection := IdentifierSchemaStruct{}
+	if plMap["privatelink_connection"] != nil && len(plMap["privatelink_connection"].([]interface{})) > 0 {
+		privatelinkConnection = GetIdentifierSchemaStruct(plMap["privatelink_connection"].([]interface{}))
+	}
+
+	privatelinkPort, ok := plMap["privatelink_connection_port"].(int)
+	if !ok {
+		return awsPrivateLinkConnection{}
+	}
```

Comment:
hmmmm if I'm reading this as if port is not an int here then you return an empty `awsPrivatelinkConnection`, don't we want an error here?

Reply:
Ah yes, this was actually a redundant check as the port is a required parameter so this would never be needed